Lib/xml/etree/ElementPath.py

```diff
@@ -253,15 +254,20 @@ def select(context, result):
                 if elem.get(key) is not None:
                     yield elem
         return select
-    if signature == "@-='":
-        # [@attribute='value']
+    if signature == "@-='" or signature == "@-!='":
+        # [@attribute='value'] or [@attribute!='value']
         key = predicate[1]
         value = predicate[-1]
         def select(context, result):
             for elem in result:
                 if elem.get(key) == value:
                     yield elem
-        return select
+        def select_negated(context, result):
+            for elem in result:
+                if elem.get(key) is not None and elem.get(key) != value:
```

Comment:
This might benefit from the walrus operator.
```suggestion
                if (attr_value := elem.get(key)) is not None and attr_value != value:
```